include/faabric/scheduler/MpiWorld.h

```diff
@@ -180,6 +180,12 @@ class MpiWorld
 
     double getWTime();
 
+    // ---------- Testing only methods ----------
```

Comment:
Although these methods are just used in tests, they're still a valid external API for this class, providing a way to query information on the internal state of the MPI world. As a result, I don't think we need to label them as test-only or have the checks in their implementation. If someone wants to remove them, they'll see they're only used in tests, and can either leave them, or rewrite the tests. Equally if someone wants to call them in other areas of the code, I think that's also ok; they shouldn't throw an exception just because they were originally written to be used in tests.